components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -21,29 +26,128 @@ import kotlin.coroutines.CoroutineContext
  * @param pinnedSitesStorage An instance of [PinnedSiteStorage], used for storing pinned sites.
  * @param historyStorage An instance of [PlacesHistoryStorage], used for retrieving top frecent
  * sites from history.
- * @param defaultTopSites A list containing a title to url pair of default top sites to be added
- * to the [PinnedSiteStorage].
+ * @param locationService An instance of [LocationService], used for retrieving the user's country
+ * region for determining the default top sites to add.
+ * @param isDefaultSiteAdded Whether or not the default sites were already added from Fenix.
  */
 class DefaultTopSitesStorage(
+    context: Context,
     private val pinnedSitesStorage: PinnedSiteStorage,
     private val historyStorage: PlacesHistoryStorage,
-    private val defaultTopSites: List<Pair<String, String>> = listOf(),
+    private val locationService: LocationService,
+    private val isDefaultSiteAdded: Boolean,
     coroutineContext: CoroutineContext = Dispatchers.IO
 ) : TopSitesStorage, Observable<TopSitesStorage.Observer> by ObserverRegistry() {
 
+    private val sharedPref = context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
     private var scope = CoroutineScope(coroutineContext)
 
     // Cache of the last retrieved top sites
     var cachedTopSites = listOf<TopSite>()
 
     init {
-        if (defaultTopSites.isNotEmpty()) {
+        if (sharedPref.getInt(MOZAC_LOCAL_DEFAULT_TOP_SITES_VERSION, 0) !=
+            sharedPref.getInt(MOZAC_DEFAULT_TOP_SITES_VERSION, 1)
+        ) {
             scope.launch {
-                pinnedSitesStorage.addAllPinnedSites(defaultTopSites, isDefault = true)
+                initializeDefaultTopSites(context)
+
+                val region = locationService.fetchRegion()
+
+                if (!isDefaultSiteAdded) {
+                    // Add default top sites for new installs.
+                    val countryCode = region?.countryCode ?: "XX"
+                    var language = LocaleManager.getCurrentLocale(context)?.toLanguageTag()
+                        ?: LocaleManager.getSystemDefault().toLanguageTag()
+                    // Modify "en" variants such as "en-ca" to "en". If the language is not "en",
+                    // default to a generic "XX".
```

Comment:
Why are we modifying the language here?

Reply:
For Pocket, we only add it as a default top site if the user's language is english (en). You can have different variants of english such as "en-ca" for Canada. So, we strip it to "en". Otherwise, we can just have a dummy language "XX" to catch all languages.